keras/engine/training.py

```diff
@@ -2163,6 +2163,10 @@ def generate_arrays_from_file(path):
                     val_data += [0.]
                 for cbk in callbacks:
                     cbk.validation_data = val_data
+            elif do_validation and val_gen:
+                for cbk in callbacks:
+                    cbk.validation_generator = validation_data
```

Comment:
Why don't you use the already existing `Callback.validation_data` here, and check if it's a generator or a tuple? I believe this would be more aligned with the way `validation_data` handles by `Model.fit`.